src/ynEIGEN/AssetRegistry.sol

```diff
@@ -0,0 +1,323 @@
+import {AccessControlUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/access/AccessControlUpgradeable.sol";
+import {ReentrancyGuardUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/utils/ReentrancyGuardUpgradeable.sol";
+import {IStrategyManager} from "lib/eigenlayer-contracts/src/contracts/interfaces/IStrategyManager.sol";
+import {IDelegationManager} from "lib/eigenlayer-contracts/src/contracts/interfaces/IDelegationManager.sol";
+import {IStrategy} from "lib/eigenlayer-contracts/src/contracts/interfaces/IStrategy.sol";
+import {IEigenStrategyManager} from "src/interfaces/IEigenStrategyManager.sol";
+import {IERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+import {IERC20Metadata} from "lib/openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import {ITokenStakingNodesManager} from "src/interfaces/ITokenStakingNodesManager.sol";
+import {IynEigen} from "src/interfaces/IynEigen.sol";
+import {IwstETH} from "src/external/lido/IwstETH.sol";
+import {IERC4626} from "lib/openzeppelin-contracts/contracts/interfaces/IERC4626.sol";
+import {IRateProvider} from "src/interfaces/IRateProvider.sol";
+import {IAssetRegistry} from "src/interfaces/IAssetRegistry.sol";
+import "lib/openzeppelin-contracts-upgradeable/contracts/access/AccessControlUpgradeable.sol";
+
+interface IAssetRegistryEvents {
+    event AssetAdded(address indexed asset);
+    event AssetDeleted(address indexed asset, uint256 assetIndex);
+    event AssetActivated(address indexed asset);
+    event AssetDeactivated(address indexed asset);
+    event PausedUpdated(bool paused);
+}
+
+/** @title AssetRegistry.sol
+ *  @dev This contract handles the strategy management for ynEigen asset allocations.
+ */
+ contract AssetRegistry is IAssetRegistry, AccessControlUpgradeable, ReentrancyGuardUpgradeable, IAssetRegistryEvents {
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  ERRORS  ------------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    error UnsupportedAsset(IERC20 asset);
+    error Paused();
+    error AssetNotActiveOrNonexistent(address inactiveAsset);
+    error AssetBalanceNonZeroInPool(uint256 balanceInPool);
+    error AssetBalanceNonZeroInStrategyManager(uint256 balanceInStrategyManager);
+    error AssetNotFound(address absentAsset);
+    error ZeroAmount();
+    error ZeroAddress();
+    error LengthMismatch(uint256 length1, uint256 length2);
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  ROLES  -------------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    bytes32 public constant ASSET_MANAGER_ROLE = keccak256("ASSET_MANAGER_ROLE");
```

Comment:
ASSET_MANAGER_ROLE Does not appear to be declared in Actors.sol, or set within the Intialize function.